contrib/hashicorp/consul/option.go

```diff
@@ -72,3 +76,10 @@ func WithAnalyticsRate(rate float64) ClientOption {
 		}
 	}
 }
+
+// WithConfig extracts the config information for the client to be tagged
+func WithConfig(config *consul.Config) ClientOption {
+	return func(cfg *clientConfig) {
+		cfg.hostname = strings.Split(config.Address, ":")[0]
```

Comment:
[nit] maybe we could use [net.SplitHostPort](https://pkg.go.dev/net#SplitHostPort) here instead.

Reply:
Done